front/src/utils/step/step.util.ts

```diff
@@ -19,128 +19,131 @@ import {
 import type { UserInfos, UserSituation } from "../../types";
 import { getObjectValue } from "./../storage.util";
 
-const getUserInfos = (
-  userSituations: UserSituation[] | null,
+export const checkErrorOnProfile = (
+  userSituation: UserSituation | undefined,
   childBirthday: string | null | undefined
-): UserInfos => {
-  const infos: UserInfos = _.cloneDeep(DEFAULT_USER_INFOS);
-  if (userSituations && userSituations.length > 0) {
-    userSituations.map((userSituation) => {
-      const id = userSituation.id as keyof typeof UserInfo;
-      infos[id] = userSituation.isChecked;
-    });
-  }
-  if (childBirthday && childBirthday.length > 0) infos.date = childBirthday;
+): string | null => {
+  const information = getUserInformation(userSituation, childBirthday);
+  if (!information.conception && !information.projet) {
+    if (!information.date) return Labels.profile.dateIsRequired;
 
-  return infos;
+    const date = new Date(information.date);
+    const now = new Date();
+    if (information.grossesse) return getPregnancyErrorMessageOrNull(now, date);
+    if (information.enfant || information.enfants)
+      return getErrorMessageForChildOrNull(now, date);
+  }
+  return null;
 };
 
-const checkErrorForGrossesse = (date: Date): string | null => {
-  let errorMessage = null;
-  const now = new Date();
-  const grossesseDebut = subWeeks(date, GROSSESSE_TOTAL_SEMAINES_SA);
-  if (isBefore(now, grossesseDebut)) {
-    errorMessage = Labels.profile.dateTooFarInFuture;
-  } else if (isBefore(date, now)) {
-    errorMessage = Labels.profile.dateCannotBeInThePast;
+export const getCheckedUserSituationOrUndefined = (
+  userSituations: UserSituation[] | null
+): UserSituation | undefined => _.find(userSituations, { isChecked: true });
+
+export const getCurrentStepIdOrNull = (
+  checkedUserSituation: UserSituation | undefined,
+  childBirthday: string | null | undefined
+): number | null => {
+  const information = getUserInformation(checkedUserSituation, childBirthday);
+  if (
```

Comment:
Il n'y a pas moyen de faire un switch ?

Reply:
ça rend ça : 

```javascript
switch (true) {
    case information.projet:
      return StepId.projet;
    case information.conception:
      return StepId.conception;
    case information.grossesse:
    case information.enfant:
    case information.enfants:
      if (information.date) {
        const date = new Date(information.date);
        return currentStepId(date);
      }
      return null;
    default:
      return null;
  }
```

Reply:
why not, t'en pense quoi ?

Reply:
ouais euh peut-être pas le bon plan en fait

Reply:
ouais je sais pas si c'est vraiment plus lisible :/ tu veux que je push ça ou je laisse comme avant ?